sparta/test/SpartaSharedPointer/SpartaSharedPointer_test.cpp

```diff
@@ -200,6 +219,180 @@ void testMoveSupport()
     EXPECT_EQUAL(ptr10.use_count(), 2);
 }
 
+void testUpcastingConversionSupport()
+{
+    sparta::SpartaSharedPointer<DerivedType> d_ptr(new DerivedType);
+    sparta::SpartaSharedPointer<MyType> b_ptr;
+
+    EXPECT_EQUAL(d_ptr.use_count(), 1);
+    EXPECT_EQUAL(b_ptr.use_count(), 0);
+
+    // Upcasting is allowed (derived -> base)
+    EXPECT_NOTHROW(b_ptr = d_ptr);
+
+    EXPECT_EQUAL(d_ptr.use_count(), 2);
+    EXPECT_EQUAL(b_ptr.use_count(), 2);
+
+    // Downcasting and other types of casting are not (will throw static assert)
+    //EXPECT_THROW(d_ptr = b_ptr);
+
+    d_ptr->a = 5;
+    EXPECT_TRUE(d_ptr->a == 5);
+    EXPECT_TRUE(b_ptr->a == 5);
+
+    d_ptr->b = 7;
+    EXPECT_TRUE(d_ptr->b == 7);
+    // error: no member named 'b' in 'MyType'
+    //EXPECT_THROW(b_ptr2->b);
+
+    // Original pointer can be destructed and copy is undisturbed
+    d_ptr = nullptr;
+    EXPECT_TRUE(b_ptr->a == 5);
+    EXPECT_EQUAL(b_ptr.use_count(), 1);
+
+    // Original pointer can be reassigned and copy is undisturbed
+    d_ptr.reset(new DerivedType);
+    d_ptr->a = 50;
+    EXPECT_TRUE(d_ptr->a == 50);
+    EXPECT_EQUAL(d_ptr.use_count(), 1);
+    EXPECT_TRUE(b_ptr->a == 5);
+    EXPECT_EQUAL(b_ptr.use_count(), 1);
+
+    // Copy can be destructed and original is undisturbed
+    b_ptr = d_ptr;
```

Comment:
Right here, I'm betting `DerivedType` won't be deleted.  `MyType` will, however...